crt_portal/cts_forms/views.py

```diff
@@ -436,31 +436,58 @@ def process_intake_filters(request):
     }
 
 
-def get_action_data(requested_actions):
+def get_action_data(requested_actions, report_url_args, paginated_offset):
     data = []
-    for action in requested_actions:
+    for index, action in enumerate(requested_actions):
         data.append({
             "action": action.verb,
             "detail": action.description,
             "timestamp": action.timestamp,
             "reportid": action.target_object_id,
-            "url": f'/form/view/{action.target_object_id}'
+            "url": f'/form/view/{action.target_object_id}?next={report_url_args}&index={paginated_offset + index}'
         })
     return data
 
 
 def process_activity_filters(request):
     query_filters, selected_actions = dashboard_filter(request.GET)
+    per_page = request.GET.get('per_page', 15)
+    page = request.GET.get('page', 1)
+    sort_expr, sorts = activity_sort(request.GET.getlist('sort'))
+    if selected_actions != []:
+        selected_actions = selected_actions.order_by(*sort_expr)
+    paginator = Paginator(selected_actions, per_page)
+    selected_actions, page_format = pagination(paginator, page, per_page)
+    sort_state = {}
+    # make sure the links for this page have the same paging, sorting, filtering etc.
+    page_args = f'?per_page={per_page}'
+    # process filter query params
+    filter_args = get_filter_args(query_filters)
+    page_args += filter_args
+
+    # process sort query params
+    sort_args, sort_state = get_sort_args(sorts, sort_state)
```

Comment:
these comments seem redundant

```suggestion
    filter_args = get_filter_args(query_filters)
    page_args += filter_args

    sort_args, sort_state = get_sort_args(sorts, sort_state)
```